### PR TITLE
Added various getters to return content from a domain designer page.

### PR DESCRIPTION
**Pull request by labkey-danield**
Made some minor changes to the DomainDesigner pages to return list of the panes present (needed to validate behavior in SampleManager).
Fixed an issue in DomainFieldRow.getType.
Added getters for some of the fields in the ReactDesignerPage that are used in the SampleManager tests.

### CODE REVIEW

File: src/org/labkey/test/components/glassLibrary/components/MultiMenu.java [hunk 1]
```diff
@@ -60,16 +62,67 @@ public void doMenuAction(List<String> pathToAction)
...
+            if(stale || loading )
+                WebDriverWrapper.sleep(500);
+
+            tries++;
+
+            // Keep trying until we get valid menu text.
+        } while((stale || loading) && (tries <= 10));
+
+        if(tries > 10)
+            throw new RuntimeException("Menu text never returned.");
```
Comment (labkey-tchad) [inline src/org/labkey/test/components/glassLibrary/components/MultiMenu.java]: Use the stale and loading flags to trigger this error state. It's less prone to off-by-one errors and we could generate more informative error messages (maybe include the menu text if we found any before running out of time). e.g.
```
if (stale)
{
    throw new RuntimeException("Menu items kept going stale");
}
if (loading)
{
    throw new RuntimeException("Menu items still loading. " + menuText.toString());
}
```

File: src/org/labkey/test/components/glassLibrary/components/MultiMenu.java [hunk 2]
```diff
@@ -60,16 +62,67 @@ public void doMenuAction(List<String> pathToAction)
...
+
+                }
+
+            }
+            catch(StaleElementReferenceException staleExc)
+            {
+                // This happens in the time between the change of the menu content from containing "loading"
+                // to having data (like "sample sets").
+                stale = true;
+            }
```
Comment (labkey-tchad) [inline src/org/labkey/test/components/glassLibrary/components/MultiMenu.java]: Make sure that it wasn't the menu itself that went stale so this doesn't keep looping futilely.
```
catch (..)
{
    [..]
    getComponentElement().isDisplayed(); // will throw an uncaught 'StaleReferenceException' if the entire menu went stale
    stale = true;
}
```

File: src/org/labkey/test/pages/ReactAssayDesignerPage.java [hunk 3]
```diff
@@ -53,12 +52,29 @@ public ReactAssayDesignerPage setName(String name)
...
+    public String getName()
+    {
+        return elementCache().nameInput.get();
+    }
+
+    // I think the name field is the only field that can be enabled/disabled.
+    // For example once an assay is created you can't change it's name so the field will be disabled.
+    public boolean isNameEnabled()
+    {
+        return null == elementCache().nameInput.getComponentElement().getAttribute( "disabled");
```
Comment (labkey-tchad) [inline src/org/labkey/test/pages/ReactAssayDesignerPage.java]: If we're using the `disabled` attribute, I think the standard `isEnabled()` method would work. (Though perhaps not)
```suggestion
        return elementCache().nameInput.getComponentElement().isEnabled();
```

File: src/org/labkey/test/components/glassLibrary/components/MultiMenu.java (hunk 1 above)
Comment (labkey-danield) [reply src/org/labkey/test/components/glassLibrary/components/MultiMenu.java]: Updated.

File: src/org/labkey/test/components/glassLibrary/components/MultiMenu.java (hunk 2 above)
Comment (labkey-danield) [reply src/org/labkey/test/components/glassLibrary/components/MultiMenu.java]: Updated.

File: src/org/labkey/test/pages/ReactAssayDesignerPage.java (hunk 3 above)
Comment (labkey-danield) [reply src/org/labkey/test/pages/ReactAssayDesignerPage.java]: Ok, we will give it a shot and see.

**Pull request merged**